std/base64.d

```diff
@@ -390,7 +390,7 @@ template Base64Impl(char Map62th, char Map63th, char Padding = '=')
      * Returns:
      *  The number of times the output range's $(D put) method was invoked.
      */
-    size_t encode(R1, R2)(in R1 source, auto ref R2 range)
+    size_t encode(R1, R2)(R1 source, auto ref R2 range)
```

Comment:
Since you have `auto s = source;` below is it necessary to remove `in`?

Reply:
`in` and `out` are basically obsolete and I want to remove them whenever possible.

Reply:
What if `R1` is a large static array?

Reply:
Ah, I see: even though the first condition is `isArray`, `ElementType` only works for dynamic arrays.

On that note the arguments should probably be changed from `(R1 source, auto ref R2 range)` to `(scope const(E)[] source, auto ref R2 range)` but that could be separate pull request.